### PR TITLE
Added OpenAlex Reader for Scientific QA

### PR DESCRIPTION
**Pull request by shauryr**
# Description

OpenAlex is the largest repository of scientific articles. This loader aims to fetch data from there API and load papers in llama_index

- [x] New Loader/Tool

# How Has This Been Tested?

I have added a demo notebook to test it in action.

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have added a library.json file if a new loader/tool was added
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

image

### CODE REVIEW

File: llama_hub/openalex/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+"""Init file."""
```
Comment (anoopshrma) [inline llama_hub/openalex/__init__.py:1]: from llama_hub.pdb.base import (
    OpenAlexReader
)

__all__ = ["OpenAlexReader"]

File: llama_hub/openalex/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# OpenAlex Reader
+
+This loader will search for papers in OpenAlex and load them in llama-index. The main advantage of using OpenAlex is that you can search the full-text for Open Access papers as well.
+
+## Usage
+
+```python
+from llama_hub.openalex_loader.base import OpenAlexReader
```
Comment (anoopshrma) [inline llama_hub/openalex/README.md:8]: After making the changes in init file, Update the readme code
`from llama_hub.openalex_loader import OpenAlexReader`

File: llama_hub/openalex/__init__.py (hunk 1 above)
Comment (shauryr) [reply llama_hub/openalex/__init__.py:1]: thanks for the review! - https://github.com/run-llama/llama-hub/pull/599/commits/7abe3fa1d7148a40b9b835bb9aadbcf2aea03609

File: llama_hub/openalex/README.md (hunk 2 above)
Comment (shauryr) [reply llama_hub/openalex/README.md]: thank you for this -   https://github.com/run-llama/llama-hub/pull/599/commits/a4bc7e288f2fd6a5f5d49fa66dcbbd2d458702d4

**Review by anoopshrma — APPROVED** [review]

**Pull request merged**